go/types/encoding_test.go

```diff
@@ -168,7 +168,7 @@ func TestRoundTrips(t *testing.T) {
 		f = math.Copysign(f, -1)
 		assertRoundTrips(Number(f))
 	}
-	floatTest := []float64{1.01, 1.001, 1.0001, 1.00001, 1.000001, 100.01, 1000.000001}
+	floatTest := []float64{1.01, 1.001, 1.0001, 1.00001, 1.000001, 100.01, 1000.000001, 122.411912027329, 0.42}
```

Comment:
should the special case numbers below (JS Max/Min safe int, machine epsilon) go here maybe instead of the xp-test? makes sense for them to be here to be sure we can encode/decode roundtrip in golang properly, but also makes sense for them to exist in xp-test to be sure we get same hash across golang vs. JS too.